internal/txprocessor/txprocessor.go

```diff
@@ -86,12 +101,26 @@ func (r TXProcessor) submitTxWithProofs(queryID uint64, block *neutrontypes.Bloc
 	if err != nil {
 		return fmt.Errorf("failed to store tx: %w", err)
 	}
-	r.dequeue <- relay.PendingSubmittedTxInfo{
-		QueryID:         queryID,
-		SubmittedTxHash: hash,
-		NeutronHash:     neutronTxHash,
-		SubmitTime:      time.Now(),
-	}
+
+	// We submit the PendingSubmittedTxInfo only after checkSubmittedTxStatusDelay to reduce the possibility of
+	// unsuccessful checks (the block is 100% not ready here yet).
+	go func() {
+		var t = time.NewTimer(r.checkSubmittedTxStatusDelay)
+		select {
+		case <-t.C:
+			submittedTxsTasksQueue <- relay.PendingSubmittedTxInfo{
+				QueryID:         queryID,
+				SubmittedTxHash: hash,
+				NeutronHash:     neutronTxHash,
+				SubmitTime:      time.Now(),
```

Comment:
this submission time 1) is not correct; 2) is not used by the submit checker. wyt we can do with it? I'd remove it cuz it misleads

Reply:
yup, i don't see any usages anywhere in the code. removed this field